encoder.py

```diff
@@ -0,0 +1,144 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Optional, List, Dict, Any, Iterable
+
+import numpy as np
+
+from jina import Executor, requests, DocumentArray
+
+import torch
+import torch.nn as nn
+import torchvision.models as models
+
+
+def _batch_generator(data: List[Any], batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i:min(i + batch_size, len(data))]
+
+
+class ImageTorchEncoder(Executor):
+    """
+    :class:`ImageTorchEncoder` encodes ``Document`` content from a ndarray,
+    potentially B x (Channel x Height x Width) into a ndarray of `B x D`.
+    Where B` is the batch size and `D` is the Dimension.
+    Internally, :class:`ImageTorchEncoder` wraps the models from `
+    `torchvision.models`.
+    https://pytorch.org/docs/stable/torchvision/models.html
+    :param model_name: the name of the model. Supported models include
+        ``resnet18``, ``alexnet``, `squeezenet1_0``,  ``vgg16``,
+        ``densenet161``, ``inception_v3``, ``googlenet``,
+        ``shufflenet_v2_x1_0``, ``mobilenet_v2``, ``resnext50_32x4d``,
+        ``wide_resnet50_2``, ``mnasnet1_0``
+    :param pool_strategy: the pooling strategy. Options are:
+        - `None`: Means that the output of the model will be the 4D tensor
+            output of the last convolutional block.
+        - `mean`: Means that global average pooling will be applied to the
+            output of the last convolutional block, and thus the output of
+            the model will be a 2D tensor.
+        - `max`: Means that global max pooling will be applied.
+    :param channel_axis: The axis of the color channel, default is 1
```

Comment:
I do not understand much this parameter. Is the axis of what? the input? the one where the model expects it to be??

Reply:
it's the axis of the input channel, the expected one is _default_channel_axis. I agree that we can do this in normalizer.
Yet, I prefer to keep _default_channel_axis since it tells us which axis of this framework should we use.